pkg/api/v1alpha1/cluster.go

```diff
@@ -740,3 +759,39 @@ func validatePodIAMConfig(clusterConfig *Cluster) error {
 	}
 	return nil
 }
+
+func validateCPUpgradeRolloutStrategy(clusterConfig *Cluster) error {
+	if clusterConfig.Spec.ControlPlaneConfiguration.UpgradeRolloutStrategy == nil {
+		fmt.Println("ControlPlaneConfiguration: UpgradeRolloutStrategy not specified in cluster config. CAPI will default to 'RollingUpdate' with maxSurge=1")
+		return nil
+	}
+
+	if clusterConfig.Spec.ControlPlaneConfiguration.UpgradeRolloutStrategy.Type != "RollingUpdate" {
+		return fmt.Errorf("ControlPlaneConfiguration: only 'RollingUpdate' supported for upgrade rollout strategy type")
+	}
+
+	if clusterConfig.Spec.ControlPlaneConfiguration.UpgradeRolloutStrategy.RollingUpdate.MaxSurge > 1 {
+		return fmt.Errorf("ControlPlaneConfiguration: maxSurge for control plane must be 0 or 1")
+	}
+
+	return nil
+}
+
+func validateMDUpgradeRolloutStrategy(clusterConfig *Cluster) error {
+	if len(clusterConfig.Spec.WorkerNodeGroupConfigurations) <= 0 {
+		return nil
+	}
+	if clusterConfig.Spec.WorkerNodeGroupConfigurations[0].UpgradeRolloutStrategy == nil {
```

Comment:
we should validate all the `WorkerNodeGroupConfigurations[*].UpgradeRolloutStrategy` not just the first one